### PR TITLE
JMS: support custom JNDI properties

### PR DESCRIPTION
**Pull request by pwielgolaski**
In some cases you could need to specific JNDI properties for JMS, at the moment you need can not do it using builder, you need to create connection factory by yourself.
This PR fix it

### CODE REVIEW

File: gatling-jms/src/main/scala/io/gatling/jms/jndi/JmsJndiConnectionFactoryBuilder.scala [hunk 1]
```diff
@@ -64,6 +68,10 @@ case class JmsJndiConnectionFactoryBuilder(
       properties.put(Context.SECURITY_CREDENTIALS, credentials.password)
     }
 
+    jndiProperties.foreach { case (k, v) =>
+      properties.put(k, v)
+    }
```
Comment (slandelle) [inline gatling-jms/src/main/scala/io/gatling/jms/jndi/JmsJndiConnectionFactoryBuilder.scala]: Add in imports `import scala.collection.JavaConverters._`

then replace with `properties.putAll(jndiProperties.asJava)`

**Comment by slandelle** [comment]
@pwielgolaski Thanks!
Could you please address comment.
Also, for completeness, could you please add a call to the new DSL method in `JmsCompileTest` and doc is `jms.rst`?

**Comment by pwielgolaski** [comment]
Thank you for your comments, I applied them and it should be good to merge.

File: gatling-jms/src/main/scala/io/gatling/jms/jndi/JmsJndiConnectionFactoryBuilder.scala [hunk 2]
```diff
@@ -37,20 +39,24 @@ case class JmsJndiConnectionFactoryBuilderUrlStep(connectionFactoryName: String)
...
     connectionFactoryName: String,
     url:                   String,
-    credentials:           Option[Credentials] = None
+    credentials:           Option[Credentials] = None,
+    properties:            Map[String, String] = Map.empty
 ) {
 
   def credentials(user: String, password: String) = copy(credentials = Some(Credentials(user, password)))
 
+  def property(key: String, value: String) = copy (properties = properties + (key -> value))
```
Comment (slandelle) [inline gatling-jms/src/main/scala/io/gatling/jms/jndi/JmsJndiConnectionFactoryBuilder.scala]: I missed this one: `updated(key, value)` would be better IMO. I'll fix when I'll cherry-pick.

Comment (pwielgolaski) [inline gatling-jms/src/main/scala/io/gatling/jms/jndi/JmsJndiConnectionFactoryBuilder.scala]: I fixed it and pushed it.

**Comment by slandelle** [comment]
cherry-picked, thanks!

**Pull request closed without merging**